# 1.Introduction to data
Obj: to understand the structure of the dataset, the column types, the presense of missing values, and the overall scope of the data

In [26]:
import pandas as pd

df = pd.read_csv("C:\\Users\\Admin\\personal-finance-analysis\\data\\personal_transactions.csv")
df.shape

(806, 6)

The dataset contains 806 rows and 6 columns - a size sufficient for a comprehensive analysis

In [27]:
df.dtypes

Date                    str
Description             str
Amount              float64
Transaction Type        str
Category                str
Account Name            str
dtype: object

The "Date" column type is defined as a string - it must be changed to datetime type during the data cleaning step  

In [41]:
df.head(10)

,Date,Description,Amount,Transaction Type,Category,Account Name
0,01/01/2018,Amazon,11.11,debit,Shopping,Platinum Card
1,01/02/2018,Mortgage Payment,1247.44,debit,Mortgage & Rent,Checking
2,01/02/2018,Thai Restaurant,24.22,debit,Restaurants,Silver Card
3,01/03/2018,Credit Card Payment,2298.09,credit,Credit Card Payment,Platinum Card
4,01/04/2018,Netflix,11.76,debit,Movies & DVDs,Platinum Card
5,01/05/2018,American Tavern,25.85,debit,Restaurants,Silver Card
6,01/06/2018,Hardware Store,18.45,debit,Home Improvement,Silver Card
7,01/08/2018,Gas Company,45.00,debit,Utilities,Checking
8,01/08/2018,Hardware Store,15.38,debit,Home Improvement,Silver Card
9,01/09/2018,Spotify,10.69,debit,Music,Platinum Card


We now have an idea of what data is included in the dataset. It’s worth noting the different types of transactions (debit/credit) and payment accounts. This will affect subsequent calculations

In [29]:
df.isnull().sum()


Date                0
Description         0
Amount              0
Transaction Type    0
Category            0
Account Name        0
dtype: int64

In [30]:
df.duplicated().sum()

np.int64(0)

No duplicates or empty rows were found, which will make working with the data much easier

In [42]:
df.describe()

,Amount
count,806.000000
mean,273.391489
std,667.630374
min,1.750000
25%,15.687500
50%,37.480000
75%,117.680000
max,9200.000000


In [43]:
df["Date"].min(), df["Date"].max()

('01/01/2018', '12/29/2018')

The dataset covers data for 20 months and 29 days

In [36]:
df["Transaction Type"].value_counts()


Transaction Type
debit     688
credit    118
Name: count, dtype: int64

In [37]:
df["Category"].unique()


<StringArray>
[              'Shopping',        'Mortgage & Rent',            'Restaurants',
    'Credit Card Payment',          'Movies & DVDs',       'Home Improvement',
              'Utilities',                  'Music',           'Mobile Phone',
             'Gas & Fuel',              'Groceries',               'Paycheck',
              'Fast Food',           'Coffee Shops',               'Internet',
                'Haircut',         'Alcohol & Bars',         'Auto Insurance',
          'Entertainment',          'Food & Dining',             'Television',
 'Electronics & Software']
Length: 22, dtype: str

In [38]:
df["Account Name"].value_counts()

Account Name
Platinum Card    366
Checking         264
Silver Card      176
Name: count, dtype: int64

Main transactions are debit transactions(688 vs 118), there are a total of 22 unique payment categories, including the “Credit Card Payment” category, which refers to an internal transfer and therefore needs to be excluded from total expenses, and finally there are 3 payment methods

What needs to be corrected during the data cleaning step:
- Convert “Date” data to datetime;
- Exclude “Credit Card Payment” transactions;
- Split debit and credit transactions into two dataframes;
- Check for abnormally large transactions.
